WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift

```diff
@@ -47,7 +59,7 @@ final class OrderDetailsDataSource: NSObject {
     /// Whether the order is eligible for card present payment.
     ///
     var isEligibleForPayment: Bool {
-        return order.datePaid == nil
+        order.status == .pending || order.status == .onHold || order.status == .failed
```

Comment:
> Could we rely on the fairly new `needsPayment` property on `Order` to determine if it's eligible for payment? That property [comes from the API](https://github.com/woocommerce/woocommerce-ios/blob/b763d2c1aed04075b2b275cf2551e7610c89c085/Networking/Networking/Model/Order.swift#L196) and the fallback matches the Core rules for whether an order needs payment.

I was unaware of these new endpoints, that definitely makes sense and should make things easier when dealing with non-core statuses. Thanks!

Reply:
I modified this on 81e6692e36ee1913d98d09a98ea585553ac56de3

As originally `isEligibleForRefund` relies on `order.paidDate = nil` [here](https://github.com/woocommerce/woocommerce-ios/pull/7277/commits/81e6692e36ee1913d98d09a98ea585553ac56de3#diff-b6999e624913fd52e3e1309e073efd298acda219ddfaa387dcc28795115a77a0L61) we encounter the same problem as before, switching statuses from pending to complete and switching back grants the order a valid `datePaid`, which makes returns `isEligibleForRefund = true` incorrectly.

Doing a `!isEligibleForPayment` check here is not enough, as the [API changes](https://github.com/woocommerce/woocommerce/blob/3611d4643791bad87a0d3e6e73e031bb80447417/plugins/woocommerce/includes/class-wc-order.php#L1520-L1523) just checks for `pending` and `failed` statuses, so adding the extra check for !order.isEditable covers all our cases for the time being.

Partially related to this: https://github.com/woocommerce/woocommerce-ios/issues/7209

Reply:
Related to the comment above, as `needsPayment` does not cover an Order with `on hold`, I modified Testing case 5 to reflect this.

Reply:
Hm, I didn't realize there was a separate issue open for investigating the use of the `needsPayment` property. That sounds like a good reason to not use it in this PR, so it can be investigated and updated separately.

Is it possible to make the required changes to the "Mark Order Complete" button without changing how we define whether an order is eligible for payment?